pyhyp/pyHyp.py

```diff
@@ -758,14 +742,204 @@ def _setOptions(self):
+        # options that might be unique per layer
+        self.hyp.hypinput.volsmoothiter = self._expand_per_layer_option("volSmoothIter", dtype=numpy.int32)
+        self.hyp.hypinput.volblend = self._expand_per_layer_option("volBlend")
+        self.hyp.hypinput.volcoef = self._expand_per_layer_option("volCoef")
+        self.hyp.hypinput.epse = self._expand_per_layer_option("epsE")
+        self.hyp.hypinput.epsi = self._expand_per_layer_option("epsI")
+        self.hyp.hypinput.theta = self._expand_per_layer_option("theta")
+        self.hyp.hypinput.splay = self._expand_per_layer_option("splay")
+        self.hyp.hypinput.splayedgeorthogonality = self._expand_per_layer_option("splayEdgeOrthogonality")
+        self.hyp.hypinput.splaycornerorthogonality = self._expand_per_layer_option("splayCornerOrthogonality")
+        self.hyp.hypinput.cornerangle = self._expand_per_layer_option("cornerangle") * numpy.pi / 180
+
+        # determine marching parameters
+        full_delta_S, march_dist, growth_ratios = self._determine_marching_parameters()
+        self.hyp.hypinput.fulldeltas = full_delta_S
+        self.hyp.hypinput.marchdist = march_dist
+        self.growth_ratios = growth_ratios
+
+        # figure out pseudo grid ratio parameters
+        pgridratio, ps0 = self._configure_pseudo_grid_parameters(growth_ratios)
+        self.hyp.hypinput.pgridratio = pgridratio
+        self.hyp.hypinput.ps0 = ps0
+
+    def _determine_marching_parameters(self):
+        # if the user specified an explicit growth-ratio, use this
+        options_growth_ratios = self.getOption("growthRatios")
+        if options_growth_ratios is not None:
+            growth_ratios = self._expand_per_layer_option("growthRatios")
+
+            if self.comm.Get_rank() == 0:
+                pyHypWarning(
+                    "The option `growthRatios` has been specified. This takes precedence over `marchDist`, `nConstantStart` and `nConstantEnd`."
+                )
+        else:
+            # no growth ratio was provided -> compute it
+            growth_ratios = self._compute_growth_ratio()
+
+        # finally compute each layers deltaS and set it in fortran
+        fullDeltaS = self._compute_delta_S(growth_ratios)
+
+        # figure out what marching distance to use
+        if options_growth_ratios is not None:
+            marchDist = numpy.sum(fullDeltaS)
+        else:
+            marchDist = self.getOption("marchDist")
+
+        # let the user know what growth-ratio march dist is used
+        if self.comm.Get_rank() == 0:
+            n_decimals = 3
+            growth_ratio_string = self.get_growth_ratio_string(growth_ratios, n_decimals)
+
+            print("#--------------------#")
+            print(f"Grid Ratio:  {growth_ratio_string}")
+            print(f"March Dist:  {round_sig(marchDist, n_decimals)}")
+
+        return fullDeltaS, marchDist, growth_ratios
+
+    def get_growth_ratio_string(self, growth_ratios=None, n_decimals=3):
```

Comment:
Add argument descriptions (same for other new functions). Also probably should make it "private" with an underscore, or is there a use case for the user here that I am missing?

Reply:
I think this depends on your definition of private/public. I understood it like this: only accessed within the class = private; gets accessed outside of the class = public.

This function is called by pyHypMulti, which does not inherit from pyHyp. Thus it is accessed from the outside and should be public.

Reply:
Forgot that its used in `pyHypMulti`, so this is fine as is.